### PR TITLE
refactor: Adjust code for some Typescript errors

### PR DESCRIPTION
**Pull request by susnux**

### CODE REVIEW

File: src/components/DownloadLimitEntry.vue [hunk 1]
```diff
@@ -74,21 +75,21 @@ export default defineComponent({
...
 
 	computed: {
 		remainingCount() {
-			return this.initialLimit - this.count
+			return (this.initialLimit ?? 0) - (this.count ?? 0)
 		},
 
 		helperText() {
-			if (this.limit > 0) {
+			if (this.limit && this.limit > 0) {
```
Comment (susnux) [inline src/components/DownloadLimitEntry.vue:92]: Could be `null`, `null` can not be compared with number (strictly in TS)

File: src/components/DownloadLimitEntry.vue [hunk 2]
```diff
@@ -129,13 +130,15 @@ export default defineComponent({
...
+		t,
+
 		handleUpdateLimit(limit: string) {
 			this.limit = Number(limit) // emitted <input> value is string so we parse it to number
 		},
 
 		async onSave() {
-			const isValid = typeof this.limit === 'number' && this.limit > 0
-			if (!isValid) {
+			if (typeof this.limit !== 'number' || this.limit <= 0) {
```
Comment (susnux) [inline src/components/DownloadLimitEntry.vue:141]: With that helper Typescript can not type check the `limit` property, so directly using it in the `if` allows Typescript to narrow down the type further down the function :)

File: src/public.ts [hunk 3]
```diff
@@ -59,7 +59,7 @@ window.addEventListener('DOMContentLoaded', () => {
 	}
 
 	// Adding double-download warning
-	const downloadButtons = document.querySelectorAll<HTMLAnchorElement>('a[href*="/download/"]') || []
+	const downloadButtons = Array.from(document.querySelectorAll<HTMLAnchorElement>('a[href*="/download/"]'))
```
Comment (susnux) [inline src/public.ts:62]: `querySelectorAll` does not return an array but a `NodeList`, so passing it to a `Set` works only if the browser likes you :sweat_smile:

**Review by susnux — COMMENTED** [review]
Some comments for easier review

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**